controllers/workloads/console/controller.go

```diff
@@ -192,6 +183,15 @@ func (r *ConsoleReconciler) Reconcile(logger logr.Logger, ctx context.Context, r
 		}
 	}
 
+	// If we have yet to set the owner reference record this as a
+	// new console request
+	if len(csl.OwnerReferences) == 0 {
+		err := r.LifecycleRecorder.ConsoleRequest(ctx, csl, authRule)
+		if err != nil {
+			logging.WithNoRecord(logger).Error(err, "failed to record event", "event", "console.request")
+		}
+	}
+
```

Comment:
Will this ever fire?
Just after where this has moved from we set the console owner, getting a new `csl` object.

```suggestion
	// If we have yet to set the owner reference record this as a
	// new console request
	if !hasOwnerReferences {
		err := r.LifecycleRecorder.ConsoleRequest(ctx, csl, authRule)
		if err != nil {
			logging.WithNoRecord(logger).Error(err, "failed to record event", "event", "console.request")
		}
	}

```

We can then capture this variable in the location this section has moved from.

```go
hasOwnerReferences := len(csl.OwnerReferences) != 0
```

This should be consistent unless we get an error on the first console request after setting the owner references, but before we reach this event emission point.
We should be aware that this, whilst unlikely, is possible if the kubernetes API should become temporarily unavialable.

Reply:
Oh yes, very well spotted!! This is definitely a bug!